h2quic/response_writer.go

```diff
@@ -86,12 +98,129 @@ func (w *responseWriter) Flush() {}
 // TODO: Implement a functional CloseNotify method.
 func (w *responseWriter) CloseNotify() <-chan bool { return make(<-chan bool) }
 
+// Should do:
+// - construct a valid HTTP2 header for a request for the file to push.
+// - open a new stream (server side)
+// - send a PUSH_PROMISE containing the streamID of the new stream and the HTTP2 header
+// - use the header to create a http request and use it in ServeHTTP(w ResponseWriter, r *Request) to serve the file to push.
+// TODO (a lot of these are taken from golang.org/x/net/http2/server.go: push(target string, opts pushOptions) )
+// - check for recursive pushes: "PUSH_PROMISE frames MUST only be sent on a peer-initiated stream."
+func (w *responseWriter) Push(target string, opts *http.PushOptions) error {
+	if w.headerStream.StreamID()%2 == 0 {
```

Comment:
The `headerStream` is always QUIC stream 3. We're checking that before calling the constructor for the `responseWriter`, so you don't need to perform any check here.
Also, I don't see why this would be a recursive push. Or am I missing something obvious here?